src/DIRAC/WorkloadManagementSystem/Agent/PilotSyncAgent.py

```diff
@@ -0,0 +1,127 @@
+""" This agent syncs CS and pilot files to a web server of your choice
+
+.. literalinclude:: ../ConfigTemplate.cfg
+  :start-after: ##BEGIN PilotSyncAgent
+  :end-before: ##END
+  :dedent: 2
+  :caption: PilotsSyncAgent options
+
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+__RCSID__ = "$Id$"
+
+import os
+import json
+import shutil
+import hashlib
+import requests
+
+from DIRAC import S_OK
+from DIRAC.Core.Base.AgentModule import AgentModule
+from DIRAC.Core.Security.Locations import getHostCertificateAndKeyLocation, getCAsLocation
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+from DIRAC.DataManagementSystem.Client.DataManager import DataManager
+from DIRAC.WorkloadManagementSystem.Utilities.PilotCStoJSONSynchronizer import PilotCStoJSONSynchronizer
+
+
+class PilotSyncAgent(AgentModule):
+  """ Syncs CS and pilot files to a web server of your choice
+  """
+
+  def __init__(self, *args, **kwargs):
+    """ c'tor
+    """
+    super(PilotSyncAgent, self).__init__(*args, **kwargs)
+
+    # This location would be enough if we are running this agent on the DIRAC web server
+    # '/opt/dirac/webRoot/www/pilot'
+    self.saveDir = ''
+    self.upload = False
+    self.uploadLocations = []
+
+  def initialize(self):
+    """ Initial settings
+    """
+    self.workingDirectory = self.am_getOption('WorkDirectory')
+    self.saveDir = self.am_getOption('SaveDir', self.saveDir)
+    self.upload = self.am_getOption('Upload', self.upload)
```

Comment:
May be instead of Upload flag just check the presence of UploadLocations option and if it is, then do upload